gammapy/maps/geom.py

```diff
@@ -904,6 +902,33 @@ def __repr__(self):
         str_ += "\tcoordsys : {}\n".format(self.coordsys)
         return str_
 
+    #TODO: this is a temporary solution until we have decided how to handle
+    # quantities uniformly. This should be called after any `MapCoord.create()`
+    # to support that users can pass quantities in any Map.xxx_by_coord() method.
+    def match_axes_units(self, geom):
+        """Match the units of the non-spatial axes to a given map geometry.
+
+        Parameters
+        ----------
+        geom : `MapGeom`
+            Map geometry with specified units per axis.
+
+        Returns
+        -------
+        coords : `MapCoord`
+            Map coord object with matched units
+        """
+        coords = {}
+
+        for name, coord in self._data.items():
+            if name in ['lon', 'lat']:
+                coords[name] = coord
+            else:
+                ax = geom.get_axis_by_name(name)
+                coords[name] = u.Quantity(coord, ax.unit, copy=False).value
```

Comment:
Wouldn't it be better to put a quantity into the coords dict with the right units?
I think that would be more in the direction of the long-term solution to have / keep units in coords?

Reply:
The `.match_axes_units()` is exactly for the use case, where we can't store the quantity in the `MapCoord` object, because the coordinates are used in subsequent computations, that don't support quantities (such as the interpolation here).

Reply:
But you could still store the coords with units and then only pass the `.value` to the interpolator later. I guess our overall strategy in Gammapy should be to keep unit info as much as possible, and only strip / re-attach it when calling into Numpy or Scipy methods that don't support it.

But here it doesn't matter much for now, suggest to put as you prefer and merge.
(there was one more inline question about dict key order above)